File: packages/java/sdk/src/main/java/dev/runfabric/HandlerContext.java
package dev.runfabric;

/**
 * Request context passed to the handler (stage, function name, request ID).
 */
public final class HandlerContext {
    private final String stage;
    private final String functionName;
    private final String requestId;

    public HandlerContext(String stage, String functionName, String requestId) {
        this.stage = stage != null ? stage : "dev";
        this.functionName = functionName;
        this.requestId = requestId;
    }

    public String getStage() {
        return stage;
    }

    public String getFunctionName() {
        return functionName;
    }

    public String getRequestId() {
        return requestId;
    }
}

File: packages/dotnet/sdk/RunFabric.Sdk/HandlerContext.cs
namespace RunFabric.Sdk;

/// <summary>Request context passed to the handler (stage, function name, request ID).</summary>
public sealed class HandlerContext
{
    public string Stage { get; }
    public string? FunctionName { get; }
    public string? RequestId { get; }

    public HandlerContext(string? stage, string? functionName, string? requestId)
    {
        Stage = string.IsNullOrEmpty(stage) ? "dev" : stage;
        FunctionName = functionName;
        RequestId = requestId;
    }
}
